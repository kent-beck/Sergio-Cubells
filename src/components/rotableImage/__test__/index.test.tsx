import { render } from "@testing-library/react";
import RotableImage from "..";

/*TODO: This feature was just for fun
Probably this don't will accepted in the features,
so I don't test everething about something that will be removed.

Enjoy the Seat Ibiza 360Âº View! ðŸ¤­
*/

describe("Renders a ", () => {
  const imageList: string[] = ["image1", "image2", "imgae3"];

  test("Rotable image", () => {
    const tree = render(<RotableImage imageList={imageList} />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  test("src first image", () => {
    render(<RotableImage imageList={imageList} />);

    const testImage = document.querySelector("img") as HTMLImageElement;

    const firstImage = imageList[0];

    expect(testImage.src).toContain(firstImage);
  });

  test("empty array", () => {
    render(<RotableImage imageList={[]} />);

    const testImage = document.querySelector("img") as HTMLImageElement;

    expect(testImage.src).toContain("");
  });
});
