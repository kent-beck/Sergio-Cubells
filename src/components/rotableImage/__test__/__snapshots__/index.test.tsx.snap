// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a  Rotable image 1`] = `
<DocumentFragment>
  <div
    class="rotableImage"
  >
    <img
      draggable="false"
      src="image1"
    />
  </div>
</DocumentFragment>
`;
